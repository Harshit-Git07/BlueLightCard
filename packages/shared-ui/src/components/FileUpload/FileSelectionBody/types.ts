export interface FileSelectionProps {
  onFilesSelected: (files: File[]) => void;
  disabled?: boolean;
  allowedFileTypes?: string[]; // e.g. ['image/png', 'image/jpeg']
  maxFileSize?: number;
}
