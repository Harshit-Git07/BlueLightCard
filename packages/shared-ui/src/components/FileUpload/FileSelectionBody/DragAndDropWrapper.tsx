import { ChangeEvent, DragEvent, FC } from 'react';
import { colours, fonts } from '../../../tailwind/theme';
import FileSelectionInputButton from './InputButton';
import { formatBytes } from '../utils/formatBytes';
import { formatFileTypes } from '../utils/formatFileTypes';
import { useDragging } from '../useDragging';
import { conditionalStrings } from '../../../utils/conditionalStrings';

type Props = {
  handleDrop: (event: DragEvent<HTMLElement>) => void;
  handleButtonClick: (event: ChangeEvent<HTMLInputElement>) => void;
  disabled: boolean;
  allowedFileTypes: string[];
  maxFileSize: number;
};

const FileSelectionDragAndDropWrapper: FC<Props> = ({
  handleDrop,
  handleButtonClick,
  disabled,
  allowedFileTypes,
  maxFileSize,
}) => {
  const { dragging, ...draggingProps } = useDragging(handleDrop);

  const getTextColourStyles = (defaultColours: string) =>
    conditionalStrings({
      [colours.textOnSurfaceDisabled]: disabled,
      [colours.textPrimaryHover]: !disabled && dragging,
      [defaultColours]: !disabled && !dragging,
    });

  const sectionStyles = conditionalStrings({
    'p-6 flex flex-col items-center gap-3 border border-dashed rounded': true,
    [colours.backgroundPrimaryContainer]: dragging,
    [colours.backgroundSurfaceContainer]: !dragging,
    [colours.borderPrimaryHover]: dragging,
    [colours.borderOnSurfaceOutline]: !dragging,
  });

  return (
    <section
      className={sectionStyles}
      {...(disabled ? {} : draggingProps)}
      aria-label="Dropzone to Upload Files"
    >
      <div className="w-full flex flex-col items-center gap-6">
        <div className="w-full flex flex-col items-center gap-3">
          <FileSelectionInputButton
            handleButtonClick={handleButtonClick}
            allowedFileTypes={allowedFileTypes}
            disabled={disabled}
          />
          <p className={`hidden laptop:block ${getTextColourStyles(colours.textOnSurface)}`}>or</p>
          <p
            className={`hidden laptop:block ${fonts.bodySemiBold} ${getTextColourStyles(colours.textOnSurface)}`}
          >
            Drag and drop in here
          </p>
        </div>
        <p
          className={`text-center ${fonts.bodySmall} ${getTextColourStyles(colours.textOnSurfaceSubtle)}`}
        >
          Max file size is {formatBytes(maxFileSize)}. Supported file types are{' '}
          {formatFileTypes(allowedFileTypes)}
        </p>
      </div>
    </section>
  );
};

export default FileSelectionDragAndDropWrapper;
