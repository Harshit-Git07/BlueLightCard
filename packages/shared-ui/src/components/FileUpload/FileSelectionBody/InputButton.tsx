import { ChangeEvent, FC } from 'react';
import { ButtonColour } from '../../Button-V2';
import { ThemeVariant } from '../../../types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faUpload } from '@fortawesome/pro-regular-svg-icons';

type Props = {
  handleButtonClick: (event: ChangeEvent<HTMLInputElement>) => void;
  allowedFileTypes: string[];
  disabled: boolean;
};

const FileSelectionInputButton: FC<Props> = ({ handleButtonClick, allowedFileTypes, disabled }) => {
  const { bg, hover, text } = ButtonColour[ThemeVariant.Primary].base;
  const backgroundStyles = disabled
    ? 'bg-button-primary-disabled-bg-colour dark:bg-button-primary-disabled-bg-colour-dark text-button-primary-disabled-label-colour dark:text-button-primary-disabled-label-colour-dark'
    : bg;
  const hoverStyles = disabled ? '' : `${hover} hover:cursor-pointer`;
  const textStyles = `${text} leading-button-label-font font-button-label-font-weight`;

  const inputId = 'Upload File';

  return (
    <label
      className={`w-full px-6 py-2 flex items-center justify-center gap-2 rounded-md ${backgroundStyles} ${hoverStyles} ${textStyles}`}
      htmlFor={inputId}
    >
      <FontAwesomeIcon icon={faUpload} />
      Choose file
      <input
        id={inputId}
        title="Choose file"
        type="file"
        accept={allowedFileTypes.join(',')}
        multiple
        className="h-0 w-0"
        onChange={handleButtonClick}
        disabled={disabled}
        tabIndex={0}
      />
    </label>
  );
};

export default FileSelectionInputButton;
