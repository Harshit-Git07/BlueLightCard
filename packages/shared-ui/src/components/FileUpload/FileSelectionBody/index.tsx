import { FC, DragEvent, ChangeEvent } from 'react';
import FileSelectionDragAndDropWrapper from './DragAndDropWrapper';

export const defaultFileTypes = ['image/png', 'image/jpeg', 'application/pdf'];
export const defaultMaxUploads = 3;
export const defaultMaxFileSize = 2e7;

export type Props = {
  onFilesSelected: (files: File[]) => void;
  disabled?: boolean;
  allowedFileTypes?: string[]; // e.g. ['image/png', 'image/jpeg']
  maxFileSize?: number;
};

const FileSelectionBody: FC<Props> = ({
  onFilesSelected,
  disabled = false,
  allowedFileTypes = defaultFileTypes,
  maxFileSize = defaultMaxFileSize,
}) => {
  const onFileDrop = (event: DragEvent<HTMLElement>) => {
    event.preventDefault();
    event.stopPropagation();

    const files = event.dataTransfer.files;
    handleFileSelection(files);
  };

  const onButtonClick = (event: ChangeEvent<HTMLInputElement>) => {
    const files = event.target.files;
    handleFileSelection(files);
  };

  const handleFileSelection = (files: FileList | null) => {
    if (files) {
      const fileArray = Array.from(files);

      if (fileArray.length > 0) {
        onFilesSelected(fileArray);
      }
    }
  };

  return (
    <FileSelectionDragAndDropWrapper
      handleDrop={onFileDrop}
      handleButtonClick={onButtonClick}
      disabled={disabled}
      allowedFileTypes={allowedFileTypes}
      maxFileSize={maxFileSize}
    />
  );
};

export default FileSelectionBody;
