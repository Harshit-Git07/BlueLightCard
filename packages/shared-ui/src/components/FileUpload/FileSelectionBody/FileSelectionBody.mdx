import { Meta, Canvas, Controls } from '@storybook/blocks';
import * as FileUploadBody from './FileSelectionBody.stories';

<Meta of={FileUploadBody} />

# FileUpload

- [Overview](#overview)
- [Props](#props)
- [Variants and States](#variants-and-states)
- [Implementation](#implementation)
- [Design](#design)

## Overview

The main selection area of the FileUpload.
This component allows a user to select a set of files,
either by them dragging those files into the drop area,
or by selecting them after clicking a button.

This component does not manage file uploads,
or the state of file selection.
It relies on its parent to manage these.
See *ExampleUploadWrapper* for an example of upload management.

The component is configurable on
which filetypes are allowed to be selected,
how many files are allowed to be selected in one go,
and how large uploaded files can be.

The component has a default and disabled state.

The text that informs a user that the area is able to receive files
is removed on tablet and mobile sizes.

## Props

<Controls />

## Variants and States

The default state shown to a user when they are selecting files.
<Canvas of={FileUploadBody.Enabled} />

The state shown when the selector is disabled.
<Canvas of={FileUploadBody.Disabled} />

## Implementation
See the *ExampleUploadWrapper* for an example of Implementation.

## Design
[Figma](https://www.figma.com/design/n2NzB4G3OTR74o3dQ4BOwm/My-Account-%26-Sign-up-Component-library?node-id=100-28667&node-type=canvas&t=l1nT8MfPdmiTrK4d-0)
