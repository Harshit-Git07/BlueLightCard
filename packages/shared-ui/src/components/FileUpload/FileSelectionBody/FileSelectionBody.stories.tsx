import { Meta, StoryFn } from '@storybook/react';
import FileSelectionBody, { defaultFileTypes, defaultMaxFileSize } from '.';

const MaxWidthWrapper: typeof FileSelectionBody = (args) => (
  <div className="max-w-[350px]">
    <FileSelectionBody {...args} />
  </div>
);

// Meta data of the component to build the story
const componentMeta: Meta<typeof FileSelectionBody> = {
  title: 'Component System/FileUpload/Selector',
  component: FileSelectionBody,
  argTypes: {
    allowedFileTypes: {
      description:
        'An array of [unique file type specifiers](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/accept#unique_file_type_specifiers) that the upload selector allows.',
    },
    disabled: {
      description: 'Blocks user interaction',
    },
    maxFileSize: {
      description: 'The max size a file can be before it can no longer be selected by the user.',
    },
  },
  parameters: {
    controls: {
      exclude: ['onFilesSelected'],
    },
  },
};

const DefaultTemplate: StoryFn<typeof FileSelectionBody> = (args) => <MaxWidthWrapper {...args} />;

export const Enabled = DefaultTemplate.bind({});
Enabled.args = {
  disabled: false,
  allowedFileTypes: defaultFileTypes,
  maxFileSize: defaultMaxFileSize,
};

export const Disabled = DefaultTemplate.bind({});
Disabled.args = {
  disabled: true,
  allowedFileTypes: defaultFileTypes,
  maxFileSize: defaultMaxFileSize,
};

export default componentMeta;
