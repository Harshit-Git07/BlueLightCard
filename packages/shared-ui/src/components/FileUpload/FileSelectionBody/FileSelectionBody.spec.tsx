import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import FileSelectionBody, { Props } from '.';
import { userEvent } from '@testing-library/user-event';

const mockFilesSelected = jest.fn();
describe('FileSelectionBody component', () => {
  const defaultProps: Props = {
    onFilesSelected: mockFilesSelected,
  };
  const pngFile = new File(['(⌐□_□)'], 'chucknorris.png', { type: 'image/png' }); // react testing lib docs
  const pdfFile = new File(['(⌐□_□)'], 'chucknorris.pdf', { type: 'application/pdf' });

  beforeEach(() => {
    jest.clearAllMocks();
  });

  // smoke test
  it('should render component without error', () => {
    const { baseElement } = render(<FileSelectionBody {...defaultProps} />);
    expect(baseElement).toBeTruthy();
  });

  it('matches snapshot', () => {
    const { container } = render(<FileSelectionBody {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot when hovering a file', () => {
    const { container } = render(<FileSelectionBody {...defaultProps} />);

    const dropzone = screen.getByLabelText('Dropzone to Upload Files');

    fireEvent.dragEnter(dropzone, {
      dataTransfer: {
        files: [pngFile],
      },
    });

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot when disabled', () => {
    const { container } = render(<FileSelectionBody {...defaultProps} disabled />);

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with different allowedFileTypes and maxFileSize', () => {
    const { container: smallProps } = render(
      <FileSelectionBody {...defaultProps} allowedFileTypes={[]} maxFileSize={0} />,
    );
    expect(smallProps).toMatchSnapshot();
    const { container: mediumProps } = render(
      <FileSelectionBody {...defaultProps} allowedFileTypes={['image/png']} maxFileSize={2000} />,
    );
    expect(mediumProps).toMatchSnapshot();
    const { container: largeProps } = render(
      <FileSelectionBody
        {...defaultProps}
        allowedFileTypes={['image/png', 'application/pdf']}
        maxFileSize={2000000000}
      />,
    );
    expect(largeProps).toMatchSnapshot();
  });

  it('triggers the callback when valid files are selected via input', async () => {
    render(<FileSelectionBody {...defaultProps} />);
    const label = screen.getByLabelText(/Choose File/i);

    await userEvent.upload(label, [pngFile, pdfFile]);

    expect(getLastCallFileNames(mockFilesSelected)).toEqual(['chucknorris.png', 'chucknorris.pdf']);
  });

  it('triggers the callback when multiple valid files are dragged in', () => {
    render(<FileSelectionBody {...defaultProps} />);

    const dropzone = screen.getByLabelText(/Dropzone/i);

    fireEvent.drop(dropzone, {
      dataTransfer: {
        files: [pngFile, pdfFile],
      },
    });

    expect(getLastCallFileNames(mockFilesSelected)).toEqual(['chucknorris.png', 'chucknorris.pdf']);
  });

  it('blocks callbacks when in disabled state', () => {
    render(<FileSelectionBody {...defaultProps} disabled />);

    const label = screen.getByLabelText(/Choose File/i);
    expect(label).toBeDisabled();

    const dropzone = screen.getByLabelText(/Dropzone/i);
    fireEvent.drop(dropzone, {
      dataTransfer: {
        files: [pngFile, pdfFile],
      },
    });
    expect(mockFilesSelected).not.toHaveBeenCalled();
  });
});

const getLastCallFileNames = (mockFn: jest.Mock) => {
  const lastCallArgs = mockFn.mock.lastCall[0];

  return lastCallArgs.map((file: File) => file.name);
};
