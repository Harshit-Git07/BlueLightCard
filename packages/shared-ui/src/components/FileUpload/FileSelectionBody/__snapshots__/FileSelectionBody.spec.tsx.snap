// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileSelectionBody component matches snapshot 1`] = `
<div>
  <section
    aria-label="Dropzone to Upload Files"
    class="p-6 flex flex-col items-center gap-3 border border-dashed rounded bg-colour-surface-container-light dark:bg-colour-surface-container-dark border-colour-onSurface-outline dark:border-colour-onSurface-outline-dark"
  >
    <div
      class="w-full flex flex-col items-center gap-6"
    >
      <div
        class="w-full flex flex-col items-center gap-3"
      >
        <label
          class="w-full px-6 py-2 flex items-center justify-center gap-2 rounded-md bg-button-primary-default-bg-colour-light dark:bg-button-primary-default-bg-colour-dark hover:bg-button-primary-hover-bg-colour-light dark:hover:bg-button-primary-hover-bg-colour-dark hover:text-button-primary-hover-label-colour-light dark:hover:text-button-primary-hover-label-colour-dark hover:cursor-pointer text-button-primary-default-label-colour-light dark:text-button-primary-default-label-colour-dark leading-button-label-font font-button-label-font-weight"
          for="Upload File"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-upload "
            data-icon="upload"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M280 360c0 13.3-10.7 24-24 24s-24-10.7-24-24V81.9l-95 95c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9L239 7c9.4-9.4 24.6-9.4 33.9 0L409 143c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0l-95-95V360zm32-8V304H448c35.3 0 64 28.7 64 64v80c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V368c0-35.3 28.7-64 64-64H200v48H64c-8.8 0-16 7.2-16 16v80c0 8.8 7.2 16 16 16H448c8.8 0 16-7.2 16-16V368c0-8.8-7.2-16-16-16H312zm72 56a24 24 0 1 1 48 0 24 24 0 1 1 -48 0z"
              fill="currentColor"
            />
          </svg>
          Choose file
          <input
            accept="image/png,image/jpeg,application/pdf"
            class="h-0 w-0"
            id="Upload File"
            multiple=""
            tabindex="0"
            title="Choose file"
            type="file"
          />
        </label>
        <p
          class="hidden laptop:block text-colour-onSurface dark:text-colour-onSurface-dark"
        >
          or
        </p>
        <p
          class="hidden laptop:block font-typography-body-semibold font-typography-body-semibold-weight text-typography-body-semibold leading-typography-body-semibold tracking-typography-body-semibold text-colour-onSurface dark:text-colour-onSurface-dark"
        >
          Drag and drop in here
        </p>
      </div>
      <p
        class="text-center font-typography-body-small font-typography-body-small-weight text-typography-body-small leading-typography-body-small tracking-typography-body-small text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark"
      >
        Max file size is 
        20MB
        . Supported file types are
         
        .PNG, .JPEG or .PDF
      </p>
    </div>
  </section>
</div>
`;

exports[`FileSelectionBody component matches snapshot when disabled 1`] = `
<div>
  <section
    aria-label="Dropzone to Upload Files"
    class="p-6 flex flex-col items-center gap-3 border border-dashed rounded bg-colour-surface-container-light dark:bg-colour-surface-container-dark border-colour-onSurface-outline dark:border-colour-onSurface-outline-dark"
  >
    <div
      class="w-full flex flex-col items-center gap-6"
    >
      <div
        class="w-full flex flex-col items-center gap-3"
      >
        <label
          class="w-full px-6 py-2 flex items-center justify-center gap-2 rounded-md bg-button-primary-disabled-bg-colour dark:bg-button-primary-disabled-bg-colour-dark text-button-primary-disabled-label-colour dark:text-button-primary-disabled-label-colour-dark  text-button-primary-default-label-colour-light dark:text-button-primary-default-label-colour-dark leading-button-label-font font-button-label-font-weight"
          for="Upload File"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-upload "
            data-icon="upload"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M280 360c0 13.3-10.7 24-24 24s-24-10.7-24-24V81.9l-95 95c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9L239 7c9.4-9.4 24.6-9.4 33.9 0L409 143c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0l-95-95V360zm32-8V304H448c35.3 0 64 28.7 64 64v80c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V368c0-35.3 28.7-64 64-64H200v48H64c-8.8 0-16 7.2-16 16v80c0 8.8 7.2 16 16 16H448c8.8 0 16-7.2 16-16V368c0-8.8-7.2-16-16-16H312zm72 56a24 24 0 1 1 48 0 24 24 0 1 1 -48 0z"
              fill="currentColor"
            />
          </svg>
          Choose file
          <input
            accept="image/png,image/jpeg,application/pdf"
            class="h-0 w-0"
            disabled=""
            id="Upload File"
            multiple=""
            tabindex="0"
            title="Choose file"
            type="file"
          />
        </label>
        <p
          class="hidden laptop:block text-colour-onSurface-disabled dark:text-colour-onSurface-disabled-dark"
        >
          or
        </p>
        <p
          class="hidden laptop:block font-typography-body-semibold font-typography-body-semibold-weight text-typography-body-semibold leading-typography-body-semibold tracking-typography-body-semibold text-colour-onSurface-disabled dark:text-colour-onSurface-disabled-dark"
        >
          Drag and drop in here
        </p>
      </div>
      <p
        class="text-center font-typography-body-small font-typography-body-small-weight text-typography-body-small leading-typography-body-small tracking-typography-body-small text-colour-onSurface-disabled dark:text-colour-onSurface-disabled-dark"
      >
        Max file size is 
        20MB
        . Supported file types are
         
        .PNG, .JPEG or .PDF
      </p>
    </div>
  </section>
</div>
`;

exports[`FileSelectionBody component matches snapshot when hovering a file 1`] = `
<div>
  <section
    aria-label="Dropzone to Upload Files"
    class="p-6 flex flex-col items-center gap-3 border border-dashed rounded bg-colour-primary-container-light dark:bg-colour-primary-container-dark border-colour-primary-hover dark:border-colour-primary-hover-dark"
  >
    <div
      class="w-full flex flex-col items-center gap-6"
    >
      <div
        class="w-full flex flex-col items-center gap-3"
      >
        <label
          class="w-full px-6 py-2 flex items-center justify-center gap-2 rounded-md bg-button-primary-default-bg-colour-light dark:bg-button-primary-default-bg-colour-dark hover:bg-button-primary-hover-bg-colour-light dark:hover:bg-button-primary-hover-bg-colour-dark hover:text-button-primary-hover-label-colour-light dark:hover:text-button-primary-hover-label-colour-dark hover:cursor-pointer text-button-primary-default-label-colour-light dark:text-button-primary-default-label-colour-dark leading-button-label-font font-button-label-font-weight"
          for="Upload File"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-upload "
            data-icon="upload"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M280 360c0 13.3-10.7 24-24 24s-24-10.7-24-24V81.9l-95 95c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9L239 7c9.4-9.4 24.6-9.4 33.9 0L409 143c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0l-95-95V360zm32-8V304H448c35.3 0 64 28.7 64 64v80c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V368c0-35.3 28.7-64 64-64H200v48H64c-8.8 0-16 7.2-16 16v80c0 8.8 7.2 16 16 16H448c8.8 0 16-7.2 16-16V368c0-8.8-7.2-16-16-16H312zm72 56a24 24 0 1 1 48 0 24 24 0 1 1 -48 0z"
              fill="currentColor"
            />
          </svg>
          Choose file
          <input
            accept="image/png,image/jpeg,application/pdf"
            class="h-0 w-0"
            id="Upload File"
            multiple=""
            tabindex="0"
            title="Choose file"
            type="file"
          />
        </label>
        <p
          class="hidden laptop:block text-colour-primary-hover dark:text-colour-primary-hover-dark"
        >
          or
        </p>
        <p
          class="hidden laptop:block font-typography-body-semibold font-typography-body-semibold-weight text-typography-body-semibold leading-typography-body-semibold tracking-typography-body-semibold text-colour-primary-hover dark:text-colour-primary-hover-dark"
        >
          Drag and drop in here
        </p>
      </div>
      <p
        class="text-center font-typography-body-small font-typography-body-small-weight text-typography-body-small leading-typography-body-small tracking-typography-body-small text-colour-primary-hover dark:text-colour-primary-hover-dark"
      >
        Max file size is 
        20MB
        . Supported file types are
         
        .PNG, .JPEG or .PDF
      </p>
    </div>
  </section>
</div>
`;

exports[`FileSelectionBody component matches snapshot with different allowedFileTypes and maxFileSize 1`] = `
<div>
  <section
    aria-label="Dropzone to Upload Files"
    class="p-6 flex flex-col items-center gap-3 border border-dashed rounded bg-colour-surface-container-light dark:bg-colour-surface-container-dark border-colour-onSurface-outline dark:border-colour-onSurface-outline-dark"
  >
    <div
      class="w-full flex flex-col items-center gap-6"
    >
      <div
        class="w-full flex flex-col items-center gap-3"
      >
        <label
          class="w-full px-6 py-2 flex items-center justify-center gap-2 rounded-md bg-button-primary-default-bg-colour-light dark:bg-button-primary-default-bg-colour-dark hover:bg-button-primary-hover-bg-colour-light dark:hover:bg-button-primary-hover-bg-colour-dark hover:text-button-primary-hover-label-colour-light dark:hover:text-button-primary-hover-label-colour-dark hover:cursor-pointer text-button-primary-default-label-colour-light dark:text-button-primary-default-label-colour-dark leading-button-label-font font-button-label-font-weight"
          for="Upload File"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-upload "
            data-icon="upload"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M280 360c0 13.3-10.7 24-24 24s-24-10.7-24-24V81.9l-95 95c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9L239 7c9.4-9.4 24.6-9.4 33.9 0L409 143c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0l-95-95V360zm32-8V304H448c35.3 0 64 28.7 64 64v80c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V368c0-35.3 28.7-64 64-64H200v48H64c-8.8 0-16 7.2-16 16v80c0 8.8 7.2 16 16 16H448c8.8 0 16-7.2 16-16V368c0-8.8-7.2-16-16-16H312zm72 56a24 24 0 1 1 48 0 24 24 0 1 1 -48 0z"
              fill="currentColor"
            />
          </svg>
          Choose file
          <input
            accept=""
            class="h-0 w-0"
            id="Upload File"
            multiple=""
            tabindex="0"
            title="Choose file"
            type="file"
          />
        </label>
        <p
          class="hidden laptop:block text-colour-onSurface dark:text-colour-onSurface-dark"
        >
          or
        </p>
        <p
          class="hidden laptop:block font-typography-body-semibold font-typography-body-semibold-weight text-typography-body-semibold leading-typography-body-semibold tracking-typography-body-semibold text-colour-onSurface dark:text-colour-onSurface-dark"
        >
          Drag and drop in here
        </p>
      </div>
      <p
        class="text-center font-typography-body-small font-typography-body-small-weight text-typography-body-small leading-typography-body-small tracking-typography-body-small text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark"
      >
        Max file size is 
        0 bytes
        . Supported file types are
         
      </p>
    </div>
  </section>
</div>
`;

exports[`FileSelectionBody component matches snapshot with different allowedFileTypes and maxFileSize 2`] = `
<div>
  <section
    aria-label="Dropzone to Upload Files"
    class="p-6 flex flex-col items-center gap-3 border border-dashed rounded bg-colour-surface-container-light dark:bg-colour-surface-container-dark border-colour-onSurface-outline dark:border-colour-onSurface-outline-dark"
  >
    <div
      class="w-full flex flex-col items-center gap-6"
    >
      <div
        class="w-full flex flex-col items-center gap-3"
      >
        <label
          class="w-full px-6 py-2 flex items-center justify-center gap-2 rounded-md bg-button-primary-default-bg-colour-light dark:bg-button-primary-default-bg-colour-dark hover:bg-button-primary-hover-bg-colour-light dark:hover:bg-button-primary-hover-bg-colour-dark hover:text-button-primary-hover-label-colour-light dark:hover:text-button-primary-hover-label-colour-dark hover:cursor-pointer text-button-primary-default-label-colour-light dark:text-button-primary-default-label-colour-dark leading-button-label-font font-button-label-font-weight"
          for="Upload File"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-upload "
            data-icon="upload"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M280 360c0 13.3-10.7 24-24 24s-24-10.7-24-24V81.9l-95 95c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9L239 7c9.4-9.4 24.6-9.4 33.9 0L409 143c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0l-95-95V360zm32-8V304H448c35.3 0 64 28.7 64 64v80c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V368c0-35.3 28.7-64 64-64H200v48H64c-8.8 0-16 7.2-16 16v80c0 8.8 7.2 16 16 16H448c8.8 0 16-7.2 16-16V368c0-8.8-7.2-16-16-16H312zm72 56a24 24 0 1 1 48 0 24 24 0 1 1 -48 0z"
              fill="currentColor"
            />
          </svg>
          Choose file
          <input
            accept="image/png"
            class="h-0 w-0"
            id="Upload File"
            multiple=""
            tabindex="0"
            title="Choose file"
            type="file"
          />
        </label>
        <p
          class="hidden laptop:block text-colour-onSurface dark:text-colour-onSurface-dark"
        >
          or
        </p>
        <p
          class="hidden laptop:block font-typography-body-semibold font-typography-body-semibold-weight text-typography-body-semibold leading-typography-body-semibold tracking-typography-body-semibold text-colour-onSurface dark:text-colour-onSurface-dark"
        >
          Drag and drop in here
        </p>
      </div>
      <p
        class="text-center font-typography-body-small font-typography-body-small-weight text-typography-body-small leading-typography-body-small tracking-typography-body-small text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark"
      >
        Max file size is 
        2KB
        . Supported file types are
         
        .PNG
      </p>
    </div>
  </section>
</div>
`;

exports[`FileSelectionBody component matches snapshot with different allowedFileTypes and maxFileSize 3`] = `
<div>
  <section
    aria-label="Dropzone to Upload Files"
    class="p-6 flex flex-col items-center gap-3 border border-dashed rounded bg-colour-surface-container-light dark:bg-colour-surface-container-dark border-colour-onSurface-outline dark:border-colour-onSurface-outline-dark"
  >
    <div
      class="w-full flex flex-col items-center gap-6"
    >
      <div
        class="w-full flex flex-col items-center gap-3"
      >
        <label
          class="w-full px-6 py-2 flex items-center justify-center gap-2 rounded-md bg-button-primary-default-bg-colour-light dark:bg-button-primary-default-bg-colour-dark hover:bg-button-primary-hover-bg-colour-light dark:hover:bg-button-primary-hover-bg-colour-dark hover:text-button-primary-hover-label-colour-light dark:hover:text-button-primary-hover-label-colour-dark hover:cursor-pointer text-button-primary-default-label-colour-light dark:text-button-primary-default-label-colour-dark leading-button-label-font font-button-label-font-weight"
          for="Upload File"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-upload "
            data-icon="upload"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M280 360c0 13.3-10.7 24-24 24s-24-10.7-24-24V81.9l-95 95c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9L239 7c9.4-9.4 24.6-9.4 33.9 0L409 143c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0l-95-95V360zm32-8V304H448c35.3 0 64 28.7 64 64v80c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V368c0-35.3 28.7-64 64-64H200v48H64c-8.8 0-16 7.2-16 16v80c0 8.8 7.2 16 16 16H448c8.8 0 16-7.2 16-16V368c0-8.8-7.2-16-16-16H312zm72 56a24 24 0 1 1 48 0 24 24 0 1 1 -48 0z"
              fill="currentColor"
            />
          </svg>
          Choose file
          <input
            accept="image/png,application/pdf"
            class="h-0 w-0"
            id="Upload File"
            multiple=""
            tabindex="0"
            title="Choose file"
            type="file"
          />
        </label>
        <p
          class="hidden laptop:block text-colour-onSurface dark:text-colour-onSurface-dark"
        >
          or
        </p>
        <p
          class="hidden laptop:block font-typography-body-semibold font-typography-body-semibold-weight text-typography-body-semibold leading-typography-body-semibold tracking-typography-body-semibold text-colour-onSurface dark:text-colour-onSurface-dark"
        >
          Drag and drop in here
        </p>
      </div>
      <p
        class="text-center font-typography-body-small font-typography-body-small-weight text-typography-body-small leading-typography-body-small tracking-typography-body-small text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark"
      >
        Max file size is 
        2GB
        . Supported file types are
         
        .PNG or .PDF
      </p>
    </div>
  </section>
</div>
`;
